restapi/src/main/java/io/stargate/web/docsapi/service/DocsApiConfiguration.java

```diff
@@ -2,21 +2,40 @@
 
 import io.stargate.web.docsapi.dao.DocumentDB;
 
+/** The configuration for the document API. */
 public interface DocsApiConfiguration {
+
+  /** Default config instance */
   DocsApiConfiguration DEFAULT = new DocsApiConfiguration() {};
 
+  /** @return Returns max allowed document page size that user can request. */
   default int getMaxPageSize() {
     return DocumentDB.MAX_PAGE_SIZE;
   }
 
-  default int getSearchPageSize() {
-    return DocumentDB.SEARCH_PAGE_SIZE;
+  /**
+   * @return Returns approximate storage page size to use when querying database, based on the
+   *     amount of documents we are searching for. We consider that in average documents have 16
+   *     fields.
+   */
+  default int getApproximateStoragePageSize(int numberOfDocuments) {
+    // TODO Should 16 be configurable
```

Comment:
Yeah, maybe this should be configurable - I've already seen use-cases where real clients have ~50 fields per doc; but I also agree that maybe we should feel this out first, it is possible that 16 will work just fine

Reply:
It's hard to make it configurable as in different collections you would have different documents.. Let's keep it like this for now..